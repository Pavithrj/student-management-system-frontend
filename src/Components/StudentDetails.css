body {
    display: flex;
    justify-content: center;
    margin-top: 10px;
    text-align: center;
}

.table-header {
    border: 1px solid #000000;
    display: flex;
    justify-content: space-around;
    width: 400px;
}

.table-content {
    border: 1px solid #000000;
    display: flex;
    justify-content: space-around;
    width: 400px;
}

.crud-buttons {
    margin-top: 10px;
    margin-bottom: 10px;
    display: flex;
    justify-content: space-between;
}