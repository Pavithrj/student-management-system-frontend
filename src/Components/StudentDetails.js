import React, { useEffect, useState } from "react";
import './StudentDetails.css';
import { useLocation, useNavigate } from 'react-router-dom';

function StudentDetails() {
    const [students, setStudents] = useState([]);
    const [isSuccesful, setIsSuccesful] = useState(true);
    const location = useLocation();
    window.mylocation = location;
    // const { isSuccesfulCreated } = location;
    const navigate = useNavigate();

    const handleCreateStudent = () => {
        navigate('/create');
    }

    const handleDeleteStudent = () => {
        navigate('/delete');
        console.log("Deleted");
    }

    useEffect(() => {
        const fetchData = async () => {
            try {
                const response = await fetch('http://localhost:4000/students');
                const jsonData = await response.json();
                setStudents(jsonData);
            } catch (error) {
                console.error(error);
            }
        };
        fetchData();
    }, []);

    useEffect(() => {
        if (location.state?.isSuccesful) {
            setIsSuccesful(true);
        } else {
            setIsSuccesful(false);
        }
    }, [location]);

    return (
        <div>
            {isSuccesful ? "Successfully Created New Student" : "Try Again"}
            <h2>Students Details:</h2>
            <div className="table-header">
                <div>Name</div>
                <div>Age</div>
                <div>RollNo</div>
            </div>
            {students.map((student) => {
                const { id, name, age, rollNo } = student;
                return (
                    <div key={id} className="table-content">
                        <div>{name}</div>
                        <div>{age}</div>
                        <div>{rollNo}</div>
                    </div>
                )
            }
            )}
            <div className="crud-buttons">
                <button onClick={handleCreateStudent}>Create</button>
                <button onClick={handleDeleteStudent}>Delete</button>
            </div>
        </div>
    );
}

export default StudentDetails;