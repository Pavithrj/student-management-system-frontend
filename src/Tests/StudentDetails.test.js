import React from "react";
import { render } from "@testing-library/react";
import StudentDetails from "../Components/StudentDetails";
import { useParams } from "react-router-dom";
import { MemoryRouter, useNavigate } from 'react-router';

jest.mock("react-router-dom", () => ({
    useParams: jest.fn(),
    useNavigate: jest.fn(),
}));

describe("StudentDetails Component", () => {
    it("should match the snapshot", () => {
        // useParams.mockReturnValue({ rollNo: "123" });

        const { asFragment } = render(
            <MemoryRouter>
                <StudentDetails />
            </MemoryRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("should render the component", () => {
        // useParams.mockReturnValue({ rollNo: "123" });

        const { getByText } = render(
            <MemoryRouter>
                <StudentDetails />
            </MemoryRouter>
        );

        expect(getByText("Student Details:")).toBeInTheDocument();
        expect(getByText("Name")).toBeInTheDocument();
        expect(getByText("Age")).toBeInTheDocument();
        expect(getByText("RollNo")).toBeInTheDocument();
    });

    it("should render without any errors", () => {
        // useParams.mockReturnValue({ rollNo: "123" });

        expect(() => {
            render(
                <MemoryRouter>
                    <StudentDetails />
                </MemoryRouter>
            );
        }).not.toThrow();
    });

    it("should display 'Loading...' text initially", () => {
        // useParams.mockReturnValue({ rollNo: "123" });

        const { queryByTestId } = render(
            <MemoryRouter>
                <StudentDetails />
            </MemoryRouter>
        );

        expect(queryByTestId("loader-text")).toBeNull();
    });

    it("should render the component", () => {
        // useParams.mockReturnValue({ rollNo: "123" });

        const navigate = jest.fn();
        const useNavigateMock = jest.fn(() => navigate);
        jest.spyOn(require("react-router-dom"), "useNavigate").mockImplementation(useNavigateMock);

        const { getByText } = render(
            <MemoryRouter>
                <StudentDetails />
            </MemoryRouter>
        );

        expect(getByText("Student Details:")).toBeInTheDocument();
        expect(getByText("Name")).toBeInTheDocument();
        expect(getByText("Age")).toBeInTheDocument();
        expect(getByText("RollNo")).toBeInTheDocument();
    });
});
