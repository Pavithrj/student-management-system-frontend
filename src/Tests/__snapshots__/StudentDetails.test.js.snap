// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentDetails Component should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Student Details:
    </h1>
    <div
      class="student-details-table-header"
    >
      <div>
        Name
      </div>
      <div>
        Age
      </div>
      <div>
        RollNo
      </div>
    </div>
    <div
      class="student-details-table-content"
    >
      <div />
      <div />
      <div />
    </div>
    <div>
      <button>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
